File: Android-Client/app/src/main/java/com/example/moviespot/data/dto/ReviewDto.kt
package com.example.moviespot.data.dto

import kotlinx.serialization.Serializable

@Serializable
data class ReviewCreateDto(
    val bookingId: Int,
    val rating: Int,
    val comment: String? = null,
    val reviewDate: String? = null // ISO-8601
)

@Serializable
data class ReviewUpdateDto(
    val id: Int,
    val rating: Int,
    val comment: String? = null,
    val reviewDate: String? = null
)

@Serializable
data class ReviewResponseDto(
    val id: Int,
    val bookingId: Int,
    val rating: Int,
    val comment: String? = null,
    val reviewDate: String,
    val createdAt: String,
    val updatedAt: String,
    val userId: Int? = null,
    val movieTitle: String? = null
)

File: Backend/MovieSpot/DTO's/ReviewDTO.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace MovieSpot.DTO_s
{
    /// <summary>
    /// DTOs for transferring review data between API and client.
    /// </summary>
    public class ReviewDTO
    {
        /// <summary>
        /// DTO for creating a new review.
        /// </summary>
        public class ReviewCreateDto
        {
            [Required(ErrorMessage = "Booking ID is required.")]
            public int BookingId { get; set; }

            [Required(ErrorMessage = "Rating is required.")]
            [Range(1, 5, ErrorMessage = "Rating must be between 1 and 5.")]
            public int Rating { get; set; }

            [MaxLength(1000, ErrorMessage = "Comment cannot exceed 1000 characters.")]
            public string? Comment { get; set; }

            public DateTime? ReviewDate { get; set; }
        }

        /// <summary>
        /// DTO for updating an existing review.
        /// </summary>
        public class ReviewUpdateDto
        {
            [Required(ErrorMessage = "Review ID is required.")]
            public int Id { get; set; }

            [Range(1, 5, ErrorMessage = "Rating must be between 1 and 5.")]
            public int Rating { get; set; }

            [MaxLength(1000, ErrorMessage = "Comment cannot exceed 1000 characters.")]
            public string? Comment { get; set; }

            public DateTime? ReviewDate { get; set; }
        }

        /// <summary>
        /// DTO for returning review data in responses.
        /// </summary>
        public class ReviewResponseDto
        {
            public int Id { get; set; }
            public int BookingId { get; set; }
            public int Rating { get; set; }
            public string? Comment { get; set; }
            public DateTime ReviewDate { get; set; }
            public DateTime CreatedAt { get; set; }
            public DateTime UpdatedAt { get; set; }
            public int? UserId { get; set; }
            public string? MovieTitle { get; set; }
        }
    }
}
